src/utils/hub-channel.js

```diff
@@ -29,6 +30,33 @@ export default class HubChannel extends EventTarget {
     return this._permissions;
   }
 
+  // Migrates this hub channel to a new phoenix channel and presence
+  async migrateToSocket(socket, params) {
+    let presenceBindings;
+
+    // Unbind presence, and then set up bindings after reconnect
+    if (this.presence) {
```

Comment:
I suppose something would have gone very wrong if we end up here and the channel doesn't have a presence instance. Is it even possible? Maybe at least `console.warn` in an else clause?

Reply:
yep this seems impossible more or less, since you won't receive the message to migrate unless you managed to get connected to the `ret` channel. so failure mode would be a busted `hub` channel with a working `ret` channel, but at that point client is hosed anyway.